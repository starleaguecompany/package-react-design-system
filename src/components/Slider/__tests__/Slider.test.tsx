import * as React from 'react';
import { render, screen } from '@testing-library/react';

import { Slider } from '../../../index';
import userEvent from '@testing-library/user-event';

describe('Slider', () => {
  test('should render correctly with no props', () => {
    render(<Slider />);

    expect(screen.getByTestId('Slider')).toMatchSnapshot();
    expect(screen.getByTestId('Slider')).toBeVisible();
  });

  test('should render correctly with attributes', () => {
    render(<Slider id="test-id" className="test-class" />);

    expect(screen.getByTestId('Slider')).toMatchSnapshot();
    expect(screen.getByTestId('Slider')).toHaveClass('test-class');
    expect(screen.getByTestId('Slider')).toHaveAttribute('id', 'test-id');
  });

  test('should render correctly with disabled prop', () => {
    const { container } = render(<Slider disabled />);
    const inner = container.querySelector('.inner');
    const track = container.querySelector('.track');

    expect(screen.getByTestId('Slider')).toMatchSnapshot();
    expect(inner).toHaveClass('disabled');
    expect(track).toHaveClass('disabled');
  });

  test('should render correctly with value prop', () => {
    const { container } = render(<Slider value={60} />);
    const handler = container.querySelector('.handler');

    expect(screen.getByTestId('Slider')).toMatchSnapshot();
    expect(handler).toHaveStyle('margin-left: 60%');
  });

  test('should execute onChange callback on slider track click', async () => {
    const onChangeCallback = jest.fn(value => value);
    render(<Slider onChange={onChangeCallback} />);
    const inner = document.querySelector('.inner');

    expect(screen.getByTestId('Slider')).toMatchSnapshot();
    userEvent.click(inner);
    expect(onChangeCallback).toBeCalledTimes(1);
  });

  test('should render correctly on external value update', () => {
    const onChangeCallback = jest.fn();
    const { rerender } = render(<Slider value={60} onChange={onChangeCallback} />);

    expect(screen.getByTestId('Slider')).toMatchSnapshot();
    rerender(<Slider value={70} onChange={onChangeCallback} />);
    expect(onChangeCallback).toBeCalledTimes(0);
  });
});
