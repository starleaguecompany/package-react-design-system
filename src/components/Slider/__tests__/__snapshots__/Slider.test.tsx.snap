// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Slider should execute onChange callback on slider track click 1`] = `
<div
  data-qa="Slider"
>
  <div
    class="inner"
  >
    <div
      class="track"
    />
    <span
      class="handler"
      style="margin-left: 0%;"
    />
  </div>
</div>
`;

exports[`Slider should render correctly on external value update 1`] = `
<div
  data-qa="Slider"
>
  <div
    class="inner"
  >
    <div
      class="track"
    />
    <span
      class="handler"
      style="margin-left: 60%;"
    />
  </div>
</div>
`;

exports[`Slider should render correctly with attributes 1`] = `
<div
  class="test-class"
  data-qa="Slider"
  id="test-id"
>
  <div
    class="inner"
  >
    <div
      class="track"
    />
    <span
      class="handler"
      style="margin-left: 0%;"
    />
  </div>
</div>
`;

exports[`Slider should render correctly with disabled prop 1`] = `
<div
  data-qa="Slider"
>
  <div
    class="inner disabled"
  >
    <div
      class="track disabled"
    />
  </div>
</div>
`;

exports[`Slider should render correctly with no props 1`] = `
<div
  data-qa="Slider"
>
  <div
    class="inner"
  >
    <div
      class="track"
    />
    <span
      class="handler"
      style="margin-left: 0%;"
    />
  </div>
</div>
`;

exports[`Slider should render correctly with value prop 1`] = `
<div
  data-qa="Slider"
>
  <div
    class="inner"
  >
    <div
      class="track"
    />
    <span
      class="handler"
      style="margin-left: 60%;"
    />
  </div>
</div>
`;
